code_challenges/code_wars/increment_string.py

```diff
@@ -1,31 +1,28 @@
+import re
+
+
 def increment_string(characters):
     if characters == "":
         return "1"
     if characters[-1].isalpha():
         return characters + "1"
-    
-    numbers = ""
-    stop = None
-    for i in range(len(characters) - 1, -1, -1):
-        if characters[i].isdigit():
-            numbers += characters[i]
-            stop = i
-        else:
-            break
-
-    characters = characters[:stop]
-    numbers = numbers[::-1]
-    length = len(numbers)
-    numbers = int(numbers)
+
+    parts = re.match(r"([a-zA-Z0-9]+[a-zA-Z])([0-9]+)", characters)
+
+    first, second = parts.groups()
+
+    characters = first
+
+    numbers = int(second)
```

Comment:
Food for thought. Typically instead of doing this:
```python
first, second = parts.groups()
characters = first
numbers = int(second)
```
You would do:
```python
characters, numbers = parts.groups()
number = int(numbers)
```
because first and second are not very descriptive.